ovos_utils/gui.py

```diff
@@ -10,53 +13,84 @@
 from ovos_utils.configuration import read_mycroft_config
 
 
-def can_display():
-    return has_screen()
+def can_display() -> bool:
+    """
+    Return true if a display is available
+    """
+    return bool(has_screen())
 
 
-def is_gui_installed():
+def is_gui_installed() -> bool:
+    """
+    Return true if a GUI application is installed
+    """
     return is_installed("mycroft-gui-app") or \
            is_installed("ovos-shell") or \
            is_installed("mycroft-embedded-shell") or \
            is_installed("plasmashell")
 
 
-def is_gui_running():
+def is_gui_running() -> bool:
+    """
+    Return true if a GUI application is running
+    """
     return is_process_running("mycroft-gui-app") or \
-           is_process_running("ovos-shell") or \
-           is_process_running("mycroft-embedded-shell") or \
-           is_process_running("plasmashell")
+        is_process_running("ovos-shell") or \
+        is_process_running("mycroft-embedded-shell") or \
+        is_process_running("plasmashell")
 
 
-def is_gui_connected(bus=None):
-    # bus api for https://github.com/MycroftAI/mycroft-core/pull/2682
-    # send "gui.status.request"
-    # receive "gui.status.request.response"
+def is_gui_connected(bus=None) -> bool:
+    """
+    Check if a GUI is connected to the MessageBus.
+    sends "gui.status.request" and waits for "gui.status.request.response"
+    @param bus: MessageBusClient to use for query
+    @return: True if GUI is connected
+    """
     response = wait_for_reply("gui.status.request",
                               "gui.status.request.response", bus=bus)
     if response:
         return response.data["connected"]
     return False
 
 
-def can_use_local_gui():
+def can_use_local_gui() -> bool:
+    """
+    Returns True if a local GUI is available to connect to
```

Comment:
connect from, local GUI connects to GUI service

Reply:
Not sure I understand, this is looking for a local GUI client that can be accessed on the bus, right?

Reply:
gui clients connect to the gui service

is_connected queries the service (is there something connected)

can_use_local checks subprocess (is there a known gui client running)

local might be disconnected, connected might not be local

also you dont access guis on the bus, the guis connect to the gui websocket only (or should... in practice some such as shell connect to both)

Reply:
This check is only looking for an installed and running application, not even checking if its connected so I guess this docstring could be more precise in any case..